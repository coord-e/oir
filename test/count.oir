# test: "$OIR_BIN"
LABEL entry
  v1 = IMM 0   # カウンタ
  v2 = IMM 10  # カウンタの最大値
  v3 = IMM 0   # 和
  JUMP loop
LABEL loop
  v4 = IMM 1
  v1 = ADD v1 v4
  v3 = ADD v3 v1
  v5 = EQUAL v1 v2
  BRANCH v5 end loop
LABEL end
  RETURN v3
