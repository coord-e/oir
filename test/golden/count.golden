# OIR entry=0, exit=2
# BB0 succs={1} preds={}
LABEL entry  # 0
  v1 = IMM 0
  v2 = IMM 10
  v3 = IMM 0
  JUMP loop  # 1
# BB1 succs={2, 1} preds={0, 1}
LABEL loop  # 1
  v4 = IMM 1
  v1 = ADD v1 v4
  v3 = ADD v3 v1
  v5 = EQUAL v1 v2
  BRANCH v5 end loop  # 2, 1
# BB2 succs={} preds={1}
LABEL end  # 2
  RETURN v3
